backend/services/implementations/classService.ts

```diff
@@ -94,6 +94,22 @@ class ClassService implements IClassService {
     return (await this.mapClassToClassDTOs(classes))[0];
   }
 
+  async getClassesByTeacherId(
+    teacherId: string,
+  ): Promise<Array<ClassResponseDTO>> {
+    let classes: Class[];
+    try {
+      classes = await MgClass.find({ teacher: { $eq: teacherId } });
+      if (!classes.length) {
```

Comment:
not sure if we necessarily need this check - we can just return [] if there's no classes